### PR TITLE
add swiftformat config file

### PR DESCRIPTION
**Pull request by tomerd**
motivation: help contributors follow formatting conventions

changes: add swiftformat file that can be used before submitting PR

### CODE REVIEW

**Comment by tomerd** [comment]
@swift-ci please test

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Review by abertelrud — APPROVED** [review]

File: .swiftformat [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# file options
+
+--swiftversion 5.0
+--exclude .build
+
+# format options
+
+--self insert
```
Comment (mattt) [inline .swiftformat:8]: @tomerd As mentioned in https://github.com/apple/swift-package-manager/pull/3019#discussion_r518020505, this setting is causing SwiftFormat to erroneously add `self.`. Until we report the bug and get a fix, should we reconsider this setting?

Comment (tomerd) [reply .swiftformat:8]: hi @mattt in other projects (SwiftNIO, etc) we actually encouraged adding `self.` to make things are very explicit. happy to open up for discussion tho. @abertelrud @neonichu  @weissi

Comment (mattt) [reply .swiftformat:8]: I have no problem at all with explicit self, and don't care too strongly. My only concern is that running SwiftFormat with this setting currently produces invalid code.

Comment (tomerd) [reply .swiftformat:8]: yes, just saw and replied in https://github.com/apple/swift-package-manager/pull/3019#discussion_r518234755

**Pull request merged**